src/python/WMCore/WMStats/DataStructs/DataCache.py

```diff
@@ -1,6 +1,61 @@
 import time
+
 from WMCore.ReqMgr.DataStructs.Request import RequestInfo, protectedLFNs
 
+def _updateFilteredResult(result, key, value, curentExcluisveList, childExclusiveList):
+
+    if key not in curentExcluisveList:
+        childKeyFlag = False
+        if len(childExclusiveList) > 0:
+            for prop, cExList in childExclusiveList[0].items():
+                if key == prop:
+                    childKeyFlag = True
+                    result[key] = filterExcludeList(value, cExList, childExclusiveList[1:])
+
+            if not childKeyFlag:
+                result[key] = value
+    return
+
+def filterExcludeList(wmstatsCache, exclude, childExclusiveList):
+    """
+
+    :param wmstatsCache: wmstats cache data
+    :param exclude: dict format of {"key": "skip"|"start", "list": list of excluding property]
+    :param childExclusiveList: list of exculding property for child level. it is chained
+           - first item is a child, second item is grand child, etc (it doesn't support multiple child
+
+    :return: filtered data of wmstatsCache
+    """
+    result = {}
+    for k, v in wmstatsCache.items():
+        # when keys are not know ahead and we have to filter next level. i.e. task name, site name as key
+        if exclude["key"] == "skip":
+            result[k] = {}
+            for sk, sv in v.items():
+                _updateFilteredResult(result[k], sk, sv, exclude["list"], childExclusiveList)
+        else:
+            _updateFilteredResult(result, k, v, exclude["list"], childExclusiveList)
+
+    return result
+
+def wmstatsFilter(wmstatsCache):
```

Comment:
What's the purpose of key=start or skip?
Why these lists have different format compared to `requestExcudeList`?

Reply:
current db structure is mixed.  There are 2 types of keys. One type is unique values as key, i.e. request name, task name.  These type of key we can't filter it since we are not filtering out specific data.  
Other keys are property name {'task', 'site', etc} which can filter.  I know 'start', 'skip' is the bad name, do you have any suggestion,